File: Week -3/Qs1.py
def string_length(src):
    length = 0

    # Traverse the string until the end
    for char in src:
        length += 1

    return length

# Example usage
my_string = "Hello, world!"
print(f"The length of the string is: {string_length(my_string)}")

# OUTPUT
# The length of the string is: 13

File: Week -3/Qs1.c
#include <stdio.h>

int stringLength(char src[]) {
    int length = 0;

    // Traverse the string until we encounter the null terminator
    while (src[length] != '\0') {
        length++;
    }

    return length;
}

int main() {
    char myString[] = "Hello, world!";
    printf("The length of the string is: %d\n", stringLength(myString));
    return 0;
}

// OUTPUT
// The length of the string is: 13